pkg/frontend/querymiddleware/model.proto

```diff
@@ -67,6 +67,13 @@ message Extent  {
   // If the response is combination of multiple queries over time, all of which had timestamp set, this is the timestamp of oldest query.
   // When merging extents and some of them have 0 query timestamp, we keep non-zero timestamp, if possible.
   int64 query_timestamp_ms = 6;
+  // Number of samples processed per step to create this Extent.
```

Comment:
Can we sort it by timestamp by contract? If so, I would document it here, given it's a design decision.